From: Peter Lemenkov <lemenkov@gmail.com>
Date: Mon, 10 Apr 2023 21:01:29 +0200
Subject: [PATCH] No more cache in FetchFromCacheMiddleware

```
=================================== FAILURES ===================================
______________ DiskCacheTests.test_cache_write_unpicklable_object ______________
[gw0] linux -- Python 3.11.3 /usr/bin/python3

self = <tests.test_djangocache.DiskCacheTests testMethod=test_cache_write_unpicklable_object>

    def test_cache_write_unpicklable_object(self):
        fetch_middleware = FetchFromCacheMiddleware(empty_response)
>       fetch_middleware.cache = cache
E       AttributeError: property 'cache' of 'FetchFromCacheMiddleware' object has no setter

tests/test_djangocache.py:873: AttributeError
=============================== warnings summary ===============================
../../../../usr/lib/python3.11/site-packages/django/conf/__init__.py:240
../../../../usr/lib/python3.11/site-packages/django/conf/__init__.py:240
  /usr/lib/python3.11/site-packages/django/conf/__init__.py:240: RemovedInDjango50Warning: The USE_L10N setting is deprecated. Starting with Django 5.0, localized formatting of data will always be enabled. For example Django will display numbers and dates using the format of the current locale.
    warnings.warn(USE_L10N_DEPRECATED_MSG, RemovedInDjango50Warning)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

---------- coverage: platform linux, python 3.11.3-final-0 -----------
Name                       Stmts   Miss Branch BrPart  Cover   Missing
----------------------------------------------------------------------
diskcache/__init__.py         14      0      0      0   100%
diskcache/cli.py               0      0      0      0   100%
diskcache/core.py            928      9    421     13    98%   234->exit, 280->exit, 364->367, 371->373, 449->456, 670-673, 1190->1192, 1202->1204, 1325->exit, 1593->1567, 1707->1682, 1878->1881, 2438, 2445-2448
diskcache/djangocache.py      99      0     14      1    99%   436->446
diskcache/fanout.py          219      0     40      0   100%
diskcache/persistent.py      283      0     98      3    99%   33->36, 147->155, 1070->exit
diskcache/recipes.py         172      0     42      0   100%
----------------------------------------------------------------------
TOTAL                       1715      9    615     17    99%

Required test coverage of 98% reached. Total coverage: 98.71%
=========================== short test summary info ============================
FAILED tests/test_djangocache.py::DiskCacheTests::test_cache_write_unpicklable_object
================== 1 failed, 244 passed, 2 warnings in 25.09s ==================
```

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/tests/test_djangocache.py b/tests/test_djangocache.py
index 5f83b81..f92a6ec 100644
--- a/tests/test_djangocache.py
+++ b/tests/test_djangocache.py
@@ -870,7 +870,6 @@ class BaseCacheTests:
 
     def test_cache_write_unpicklable_object(self):
         fetch_middleware = FetchFromCacheMiddleware(empty_response)
-        fetch_middleware.cache = cache
 
         request = self.factory.get('/cache/test')
         request._cache_update_cache = True
@@ -887,7 +886,6 @@ class BaseCacheTests:
             return response
 
         update_middleware = UpdateCacheMiddleware(get_response)
-        update_middleware.cache = cache
         response = update_middleware(request)
 
         get_cache_data = fetch_middleware.process_request(request)
